website/pages/docs/from-1.5/special-variables.mdx

```diff
@@ -0,0 +1,63 @@
+---
+layout: docs
+page_title: Special Variables - HCL Configuration Language
+sidebar_title: Special Variables
+description: |-
+  Special variables provide connection information and basic instance state information.
+  This page covers all existing special variables.
+---
+
+# Build Variables
+
+`@include 'from-1.5/beta-hcl2-note.mdx'`
+
+Build variables will allow you to access, from provisioners and post-processors, connection information and basic instance state information.
+All special build variables are prefixed with `build`:
+
+```hcl
+  provisioner "shell" {
+      environment_vars = ["TESTVAR=${build.PackerRunUUID}"]
+      inline = ["echo $TESTVAR"]
+  }
+```
+
+The valid build variables to request are:
+
+- **ID**: Represents the vm being provisioned. For example, in Amazon it is the instance id; in digitalocean,
+  it is the droplet id; in Vmware, it is the vm name.
+
+- **Host**, **Port**, **User** and **Password**: The host, port, user, and password that Packer uses to access the machine.
+  Useful for using the shell local provisioner to run Ansible or Inspec against the provisioned instance.
+
+- **ConnType**: Type of communicator being used. For example, for SSH communicator this will be "ssh".
+
+- **PackerRunUUID**: Current build's unique id. Can be used to specify build artifacts.
+
+- **PackerHTTPIP**, **PackerHTTPPort**, and **PackerHTTPAddr**: HTTP IP, port, and address of the file server Packer creates to serve items in the "http" dir to the vm. The HTTP address is displayed in the format `IP:PORT`.
```

Comment:
(that's probably for the future) It would be neat if those were:
```suggestion
- **packer.http.ip**, **packer.http.port**, and **packer.http.addr**: HTTP IP, port, and address of the file server Packer creates to serve items in the "http" dir to the vm. The HTTP address is displayed in the format `IP:PORT`.
```

Reply:
Correct me if I'm wrong but in order to achieve this packer and http would have to be their own types, right?

Reply:
em it would be a specific go-cty object:

```go
map[string]cty.Value{
  "packer": cty.ObjectVal(map[string]cty.Value{
    "http": cty.ObjectVal(map[string]cty.Value{
      "ip": cty.StringVal("ipvalue),
      // etc.
    },
  },
}
```

Reply:
“Object” that’s the word. I was leaning towards block and end on type 😃. Thanks for the example super helpful.